File: Sample/NendSDK_Sample/NativeAdCarouselView.m
//
//  NativeAdCarouselView.m
//  NendSDK_Sample
//
//  Copyright © 2016年 F@N Communications. All rights reserved.
//

#import "NativeAdCarouselView.h"

#define cellWidth   [UIScreen mainScreen].bounds.size.width

static const float adPortraitWidth = 320.f; // 縦向き　広告横幅
static const float adPortraitHeight = 325.f; // 縦向き 広告高さ
static const float adLandscapeWidth = 580.f; // 横向き　広告横幅
static const float adLandscapeHeight = 200.f; // 横向き　広告高さ

@interface NativeAdCarouselView ()<NADNativeViewRendering>

@property (nonatomic, weak) IBOutlet UIImageView *nativeAdLogoImageView;
@property (nonatomic, weak) IBOutlet UILabel *nativeAdPromotionNameLabel;
@property (nonatomic, weak) IBOutlet UILabel *nativeAdPrTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *nativeAdLongTextLabel;
@property (nonatomic, weak) IBOutlet UIImageView *nativeAdImageView;
@property (nonatomic, weak) IBOutlet UILabel *nativeAdShortTextLabel;
@property (nonatomic, weak) IBOutlet UILabel *nativeAdPromotionUrlLabel;
@property (nonatomic, weak) IBOutlet UILabel *nativeAdActionButtonTextLabel;

@property (nonatomic) NSNumber *direction;

@end

@implementation NativeAdCarouselView

- (void)awakeFromNib
{
    [super awakeFromNib];
    
    _nativeAdPromotionNameLabel.adjustsFontSizeToFitWidth = YES;
    _nativeAdPromotionNameLabel.minimumScaleFactor = 0.5f;
    _nativeAdPrTextLabel.adjustsFontSizeToFitWidth = YES;
    _nativeAdPrTextLabel.minimumScaleFactor = 0.5f;
    _nativeAdLongTextLabel.adjustsFontSizeToFitWidth = YES;
    _nativeAdLongTextLabel.minimumScaleFactor = 0.5f;
    _nativeAdShortTextLabel.adjustsFontSizeToFitWidth = YES;
    _nativeAdShortTextLabel.minimumScaleFactor = 0.5f;
    _nativeAdPromotionUrlLabel.adjustsFontSizeToFitWidth = YES;
    _nativeAdPromotionUrlLabel.minimumScaleFactor = 0.5f;
    _nativeAdActionButtonTextLabel.adjustsFontSizeToFitWidth = YES;
    _nativeAdActionButtonTextLabel.minimumScaleFactor = 0.5f;
    _nativeAdActionButtonTextLabel.layer.borderColor = [UIColor blueColor].CGColor;
    _nativeAdActionButtonTextLabel.layer.borderWidth = 1.f;
    
    self.layer.borderWidth = 0.f;
}

- (void) frameUpdate:direction {
    self.direction = direction;
    [self layoutSubviews];
}

- (void) layoutSubviews {
    [super layoutSubviews];
    
    if ((self.direction).intValue == 1) {
        self.frame = CGRectMake(self.index * adPortraitWidth, 0, adPortraitWidth, adPortraitHeight);
    } else if ((self.direction).intValue == 2) {
        if (cellWidth < adLandscapeWidth) {
            // iphone4の場合、横幅設定
            self.frame = CGRectMake(self.index * (cellWidth - 20.f), 0, (cellWidth - 20.f), adLandscapeHeight);
        } else {
            self.frame = CGRectMake(self.index * adLandscapeWidth, 0, adLandscapeWidth, adLandscapeHeight);
        }
    }
    
}

#pragma mark - NADNativeViewRendering

- (UILabel *)prTextLabel
{
    return self.nativeAdPrTextLabel;
}

- (UILabel *)shortTextLabel
{
    return self.nativeAdShortTextLabel;
}

- (UILabel *)longTextLabel
{
    return self.nativeAdLongTextLabel;
}

- (UILabel *)promotionNameLabel
{
    return self.nativeAdPromotionNameLabel;
}

- (UILabel *)promotionUrlLabel
{
    return self.nativeAdPromotionUrlLabel;
}

- (UILabel *)actionButtonTextLabel
{
    return self.nativeAdActionButtonTextLabel;
}

- (UIImageView *)adImageView
{
    return self.nativeAdImageView;
}

- (UIImageView *)logoImageView
{
    return self.nativeAdLogoImageView;
}

@end

File: Sample/NendSDK_Swift_Sample/NativeAdCarouselView.swift
//
//  NativeAdCarouselView.swift
//  NendSDK_Sample
//
//  Copyright © 2016年 F@N Communications. All rights reserved.
//

import UIKit

class NativeAdCarouselView: UIView, NADNativeViewRendering {
    
    @IBOutlet private weak var nativeAdPrTextLabel: UILabel!
    @IBOutlet private weak var nativeAdShortTextLabel: UILabel!
    @IBOutlet private weak var nativeAdLongTextLabel: UILabel!
    @IBOutlet private weak var nativeAdPromotionNameLabel: UILabel!
    @IBOutlet private weak var nativeAdPromotionUrlLabel: UILabel!
    @IBOutlet private weak var nativeAdActionButtonTextLabel: UILabel!
    @IBOutlet private weak var nativeAdImageView: UIImageView!
    @IBOutlet private weak var nativeAdLogoImageView: UIImageView!
    
    internal var index: Int = 0
    private var direction: Int = 0
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        self.nativeAdPromotionNameLabel.adjustsFontSizeToFitWidth = true
        self.nativeAdPromotionNameLabel.minimumScaleFactor = 0.5
        self.nativeAdPrTextLabel.adjustsFontSizeToFitWidth = true
        self.nativeAdLongTextLabel.adjustsFontSizeToFitWidth = true
        self.nativeAdLongTextLabel.minimumScaleFactor = 0.5
        self.nativeAdShortTextLabel.adjustsFontSizeToFitWidth = true
        self.nativeAdShortTextLabel.minimumScaleFactor = 0.5
        self.nativeAdPromotionUrlLabel.adjustsFontSizeToFitWidth = true
        self.nativeAdPromotionUrlLabel.minimumScaleFactor = 0.5
        self.nativeAdActionButtonTextLabel.adjustsFontSizeToFitWidth = true
        self.nativeAdActionButtonTextLabel.minimumScaleFactor = 0.5
        self.nativeAdActionButtonTextLabel.layer.borderColor = UIColor.blueColor().CGColor
        self.nativeAdActionButtonTextLabel.layer.borderWidth = 1.0
        
        self.layer.borderWidth = 0
    }
    
    func frameUpdate(direction: Int) {
        self.direction = direction
        self.layoutSubviews()
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        if self.direction == 1 {
            self.frame = CGRectMake(CGFloat(self.index) * adPortraitWidth, 0, adPortraitWidth, adPortraitHeight)
        } else if self.direction == 2 {
            let cellWidth: CGFloat = UIScreen.mainScreen().bounds.size.width
            if cellWidth < adLandscapeWidth {
                // iphone4の場合、横幅設定
                self.frame = CGRectMake(CGFloat(self.index) * (cellWidth - 20.0), 0, (cellWidth - 20.0), adLandscapeHeight)
            } else {
                self.frame = CGRectMake(CGFloat(self.index) * adLandscapeWidth, 0, adLandscapeWidth, adLandscapeHeight)
            }
        }
    }
    
    // MARK: - NADNativeViewRendering
    
    func prTextLabel() -> UILabel! {
        return self.nativeAdPrTextLabel
    }
    
    func shortTextLabel() -> UILabel! {
        return self.nativeAdShortTextLabel
    }
    
    func longTextLabel() -> UILabel! {
        return self.nativeAdLongTextLabel
    }
    
    func promotionNameLabel() -> UILabel! {
        return self.nativeAdPromotionNameLabel
    }
    
    func promotionUrlLabel() -> UILabel! {
        return self.nativeAdPromotionUrlLabel
    }
    
    func actionButtonTextLabel() -> UILabel! {
        return self.nativeAdActionButtonTextLabel
    }
    
    func adImageView() -> UIImageView! {
        return self.nativeAdImageView
    }
    
    func logoImageView() -> UIImageView! {
        return self.nativeAdLogoImageView
    }
}
